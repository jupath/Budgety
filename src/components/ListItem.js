import React from 'react';

const ListItem = () => (
  <div>
  ListItem
  </div>
);

export default ListItem;
