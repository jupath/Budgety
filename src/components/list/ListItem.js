import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const ListItem = props => (
  <Link to={`/edit/${props.listItem.id}`}>
    <div className="list__item">
      <div>
        {props.listItem.description}<br />
        {moment(props.listItem.date).format('MMMM Do, YYYY')}
      </div>
      <div>
        {props.listItem.amount}
      </div>
    </div>
  </Link>
);

ListItem.propTypes = {
  listItem: PropTypes.shape({
    itemType: PropTypes.string,
    description: PropTypes.string,
    amount: PropTypes.number,
    date: PropTypes.number,
    note: PropTypes.string,
    id: PropTypes.string,
  }).isRequired,
};

export default ListItem;
