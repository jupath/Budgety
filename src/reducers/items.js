export default (state = [], action) => {
  switch (action.type) {
    case 'ADD_ITEM':
      return [...state, action.item];
    case 'UPDATE_ITEM':
      return state.map((item) => {
        if (item.id === action.id) {
          return {
            ...item,
            ...action.item,
          };
        }
        return item;
      });
    case 'DELETE_ITEM':
      return state.filter(item => item.id !== action.id);
    default:
      return state;
  }
};
