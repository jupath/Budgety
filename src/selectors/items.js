import moment from 'moment';

export default (items, itemsType, { startDate, endDate, sortBy }) => items.filter((item) => {
  const date = moment(item.date);
  const startDateMatch = startDate ? startDate.isSameOrBefore(date, 'day') : true;
  const endDateMatch = endDate ? endDate.isSameOrAfter(date, 'day') : true;
  const itemTypeMatch = item.itemType === itemsType;

  return startDateMatch && endDateMatch && itemTypeMatch;
}).sort((a, b) => {
  if (sortBy === 'date') {
    return a.date < b.date ? 1 : -1;
  } else if (sortBy === 'value') {
    return a.amount < b.amount ? 1 : -1;
  }
});
