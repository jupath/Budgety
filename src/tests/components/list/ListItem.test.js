import React from 'react';
import { shallow } from 'enzyme';
import items from '../../fixtures/items';
import ListItem from '../../../components/list/ListItem';

test('should render ListItem correctly', () => {
  const wrapper = shallow(<ListItem listItem={items[0]} />);
  expect(wrapper).toMatchSnapshot();
});