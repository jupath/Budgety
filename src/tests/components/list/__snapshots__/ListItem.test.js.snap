// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem correctly 1`] = `
<div
  className="listitem"
>
  <Link
    replace={false}
    to="/edit/abc123"
  >
    <div
      className="listitem__body py-3 px-2"
    >
      <div>
        Test description 1
        <br />
        <small>
          January 1st, 1970
        </small>
      </div>
      <div>
        <NumberFormat
          allowEmptyFormatting={false}
          allowNegative={true}
          decimalScale={2}
          decimalSeparator="."
          displayType="text"
          fixedDecimalScale={true}
          getInputRef={[Function]}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix="$"
          suffix=""
          thousandSeparator={true}
          type="text"
          value={3100}
        />
      </div>
    </div>
  </Link>
</div>
`;
